main.go

```diff
@@ -52,10 +55,19 @@ func main() {
 	}
 	log.Println("Received the first connection")
 
-	wg.Wait()
-	log.Println("Timed out waiting for re-connection")
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)
```

Comment:
for the future refactorings - maybe would be better to just use context? https://pkg.go.dev/os/signal#NotifyContext

that would allow to just close the context instead of pushing the "fake" SIGTERM in the goroutine

Reply:
nice article https://rudderstack.com/blog/implementing-graceful-shutdown-in-go/